lib/IRGen/IRGenSIL.cpp

```diff
@@ -8414,27 +8415,17 @@ void IRGenSILFunction::visitObjCSuperMethodInst(swift::ObjCSuperMethodInst *i) {
                               /*startAtSuper=*/true);
 }
 
-void IRGenSILFunction::visitClassMethodInst(swift::ClassMethodInst *i) {
-  assert(!i->getMember().isForeign);
-
-  Explosion base = getLoweredExplosion(i->getOperand());
-  llvm::Value *baseValue = base.claimNext();
-
-  SILDeclRef method = i->getMember().getOverriddenVTableEntry();
-  PrettyStackTraceSILDeclRef entry("lowering class method call to", method);
-
-  auto methodType = i->getType().castTo<SILFunctionType>();
-
+bool IRGenSILFunction::shouldUseDispatchThunk(SILDeclRef method) {
   AccessLevel methodAccess = method.getDecl()->getEffectiveAccess();
   auto *classDecl = cast<ClassDecl>(method.getDecl()->getDeclContext());
   bool shouldUseDispatchThunk = false;
   // Because typechecking for the debugger has more lax rules, check the access
   // level of the getter to decide whether to use a dispatch thunk for the
   // debugger.
   bool shouldUseDispatchThunkIfInDebugger =
-      !classDecl->getASTContext().LangOpts.DebuggerSupport ||
-      methodAccess == AccessLevel::Public;
-  if (IGM.hasResilientMetadata(classDecl, ResilienceExpansion::Maximal) &&
+      classDecl->getASTContext().LangOpts.DebuggerSupport &&
+      methodAccess >= AccessLevel::Public;
+  if (IGM.hasResilientMetadata(classDecl, ResilienceExpansion::Maximal) ||
```

Comment:
From my understanding changing this condition to an "or" will make expression evaluation fail in LLDB if there is no resilient metadata but `shouldUseDispatchThunkIfInDebugger` is true, since LLDB will try to call a dispatch thunk that doesn't exist. Am I mistaken and this still works correctly? This should be very trivial to test, just call an open method from LLDB.

Reply:
Yes, this reproduces:
```open class A {
    open func foo() {
        print("Here")
    }
}

func f() {
    let a  = A()
    print(a) // break
}
f()
```
On lldb:
```
bin/lldb -o 'b /break/' -o r -o 'expr a.foo()' r
error: Couldn't look up symbols:
  dispatch thunk of r.A.foo() -> ()
```

Reply:
So originally, it was
```
shouldUseDispatchThunk = ResilienceExpansion::Maximal
```
then you changed it to
```
+  bool shouldUseDispatchThunkIfInDebugger =
+      !classDecl->getASTContext().LangOpts.DebuggerSupport ||
+      methodAccess == AccessLevel::Public;
+  if (IGM.hasResilientMetadata(classDecl, ResilienceExpansion::Maximal) &&
+      shouldUseDispatchThunkIfInDebugger) {
```
which means:

DebuggerSupport = false:
ResilienceExpansion::Maximal

DebuggerSupport = true
ResilienceExpansion::Maximal && AccessLevel::Public
```

which, you're right, is not what my patch is doing.

Reply:
(I tested both on my crash reproducer and your example)